File: servidor/models/usuario.py
from sqlalchemy import Column, BigInteger, Text, Boolean, ForeignKey
from database_dos import Base

class Usuario(Base):
    __tablename__ = 'usuarios'

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    nombre_usuario = Column(Text, unique=True, nullable=False)
    contrasena = Column(Text, nullable=False)
    tienda_id = Column(Text, ForeignKey('tiendas.codigo', ondelete='SET NULL'))
    nombre = Column(Text)
    apellido = Column(Text)
    habilitado = Column(Boolean, default=True)

File: catalogos/src/main/java/com/example/demo/models/Usuario.java
package com.example.demo.models;

import jakarta.persistence.*;
import lombok.Data;


@Entity
@Table(name = "usuarios")
@Data
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "nombre_usuario", unique = true, nullable = false)
    private String nombreUsuario;

    @Column(name = "contrasena", nullable = false)
    private String contrasena;

    @ManyToOne
    @JoinColumn(name = "tienda_id", referencedColumnName = "codigo", nullable = true)
    private Tienda tienda;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "apellido")
    private String apellido;

    @Column(name = "habilitado", nullable = false)
    private boolean habilitado = true;
}

